stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/QueryChannelsController.kt

```diff
@@ -59,6 +59,9 @@ public class QueryChannelsController internal constructor(
     ) : this(filter, sort, client, domain as ChatDomainImpl)
 
     private var channelOffset = INITIAL_CHANNEL_OFFSET
+    internal var isFirstRequest: Boolean = true
```

Comment:
I wouldn't add this new state since it makes current spaghetti logic more complicated. Can we solve this issue without it on UI side? In VM for example?
Also one possible solution that we shouldn't trigger any new load more request in offline mode (until we got online result of previous request)

Reply:
The controller is the one changing the value of `isFirstRequest`, then we would create a bond between the two because ViewModel would need to update this field when update the channel list. We would also need to track `channelOffset` and `isFirstRequest` in two different places... that would also be complicated. 

Making online data wait before fetching offline data wouldn't solve the problem =|. The problem is that:
 `channelOffset` is only updated when more online data is successfully fetched. That may or may not happen, as the request can fail. If the request fails, we pick channels from the database and populate our channels... in this case, `channelOffset` is still 0, but our channel list is populated. Then the next request is treated as it was the first-page request because it is the first page **online** request.

In resume, waiting for the online that doesn't work because the request can fail =|.

Reply:
I didn't mean expose some additional data from `QueryChannelsController` to VM. I think your issue can be solved easily in UI part only. I prepared this PR #2404. As I understood your problem it will solve. Can you have a look and test it a little bit?